lib/commands/nw-test/index.js

```diff
@@ -97,8 +129,83 @@ module.exports = {
 
     var testTask = new TestTask(this.taskOptions());
 
-    return testTask.run({
-      outputPath: options.outputPath
+    return this.setNwVisibility(true, options.outputPath)
+      .then(function() {
+        return testTask.run({
+          outputPath: options.outputPath
+        });
+      });
+  },
+
+  setNwVisibility: function(visibile, outputPath) {
+    var jsonPath = path.join(outputPath, 'tests', 'package.json');
+    return readFile(jsonPath, { encoding: 'utf8' })
+    .then(function(data) {
+      var json = JSON.parse(data);
+      json.window.show = visibile;
+
+      return writeFile(jsonPath, JSON.stringify(json));
+    });
+  },
+
+  runTestsServer: function(options) {
+    var _this = this;
+    var BaseTestTask = this.tasks.TestServer;
+    var TestTask = BaseTestTask.extend({
+      testemOptions: function() {
+        /* jscs:disable requireCamelCaseOrUpperCaseIdentifiers */
+        return {
+          cwd: options.outputPath,
+          middleware: this.addonMiddlewares(),
+          reporter: require('./reporter'),
+          hide_stderr: false,
+          hide_stdout: false
+        };
+      }
+    });
+
+    var testTask = new TestTask(this.taskOptions());
+    var ui = this.ui;
+    var Watcher = require('ember-cli/lib/models/watcher');
+
+    var builder = new this.TestServerBuilder({
+        ui: ui,
+        project: this.project,
+        environment: options.environment,
+        outputPath: options.outputPath
+      });
+
+    return builder.build().then(function () {
+      return _this.setNwVisibility(true, options.outputPath);
+    })
+    .then(function() {
+      var findNW = require('../../helpers/find-nw');
+      var nwPath = safePath(findNW(_this.project));
+      var testPath = safePath(path.join(options.outputPath, 'tests'));
+      exec(nwPath + ' ' + testPath, { env: { "SERVER_MODE": "TRUE" } });
+
+      var watcher = new Watcher({
+        ui: ui,
+        builder: builder,
+        analytics: _this.analytics,
+        options: options
+      });
+
+      var watchDirOrFiles = fs.readdirSync(_this.project.root);
```

Comment:
I thought this, too since it has access to the builder.but if I comment out the `addWatchDir` call no watching occurs. Does that reproduce for you as well?